packages/synapse-react-client/src/components/SynapseTable/SynapseTableRenderers.tsx

```diff
@@ -179,10 +180,30 @@ const getEntityOrRowVersion = (
 }
 
 function AccessCell(props: CellContext<Row, unknown>) {
+  const { ref, inView } = useInView({ triggerOnce: true })
   const entityId = getEntityOrRowId(props)!
+  const hideAccessIconForExternalFileHandle =
+    props.table.options.meta?.hideAccessIconForExternalFileHandle
+  // If hideAccessIconForExternalFileHandle is true, defer rendering until the cell is in view to avoid unnecessary API calls
+  const canRenderAccessIcon =
+    !hideAccessIconForExternalFileHandle ||
+    (hideAccessIconForExternalFileHandle && inView)
+
   return (
-    <div data-testid={'AccessCell'}>
-      <HasAccessV2 key={entityId} entityId={entityId} showButtonText={false} />
+    <div
+      ref={hideAccessIconForExternalFileHandle ? ref : undefined}
```

Comment:
I think it's ok to always attach the ref, since you have `canRenderAccessIcon`
```suggestion
      ref={ref }
```